### ISSUE TITLE
[FX] optimization.fuse changes a reused Conv that is BatchNorm'd on only one path

### ISSUE THREAD

**Issue by MBUYt0n** [pytorch/pytorch#197539]
### 🐛 Describe the bug

`torch.fx.experimental.optimization.fuse` folds BN into the Conv module. If that Conv is also called on another input without BN, the second call starts using the fused weights. That is not floating-point noise: the conv+bn path matches eager, the conv-only path does not.

```python
import torch
import torch.nn as nn
from torch.fx.experimental.optimization import fuse

torch.manual_seed(0)

class M(nn.Module):
    def __init__(self):
        super().__init__()
        self.conv = nn.Conv2d(3, 8, 3, padding=1)
        self.bn = nn.BatchNorm2d(8)

    def forward(self, x, y):
        return self.bn(self.conv(x)), self.conv(y)

m = M().eval()
with torch.no_grad():
    m.bn.running_mean.fill_(0.5)
    m.bn.running_var.fill_(2.0)
    m.bn.weight.fill_(1.3)
    m.bn.bias.fill_(-0.2)

x = torch.randn(2, 3, 8, 8)
y = torch.randn(2, 3, 8, 8)

with torch.no_grad():
    eager = m(x, y)
    fused = fuse(m)
    out = fused(x, y)

print((out[0] - eager[0]).abs().max())
print((out[1] - eager[1]).abs().max())
print(fused.graph)
```

Observed on CPU nightly:

```
tensor(4.7684e-07)
tensor(0.8417)
graph():
    %x : [num_users=1] = placeholder[target=x]
    %y : [num_users=1] = placeholder[target=y]
    %conv : [num_users=1] = call_module[target=conv](args = (%x,), kwargs = {})
    %conv_1 : [num_users=1] = call_module[target=conv](args = (%y,), kwargs = {})
    return (conv, conv_1)
```

Expected: both outputs match eager. After fuse, both `call_module` nodes target the same fused `conv`, so the path that never went through BN is wrong.

`fuse()` only skips when that one conv node has multiple users. Two separate calls of the same module are two nodes, so the guard does not fire.

### Versions

```PyTorch version: 2.15.0.dev20260917

OS: macOS 26.6.2 (arm64)

Python version: 3.14.5 (main, May 10 2026, 10:21:34) [Clang 21.0.0 (clang-2100.0.123.102)] (64-bit runtime)

Python platform: macOS-26.6.2-arm64-arm-64bit-Mach-O

CPU: Apple M4 Pro```

**Comment by MBUYt0n** [comment]
I'd like to take this. Please assign it to me.

The `users > 1` check only looks at that one conv node. Two `call_module`s of the same Conv are two nodes, so `replace_node_module` still overwrites the shared submodule.

I'd skip fusion when the conv/linear target has any other `call_module` in the graph, same idea as inductor `fuse_conv_bn` already disabling fusion when one conv feeds two BNs. Then add a CPU test for the two-input shared-conv model so both outputs still match eager.

Happy to switch to a uniquely named fused conv for only the conv→bn edge if you'd rather keep that fusion.
